### PR TITLE
v1.8 backports 2020-12-08

### PR DESCRIPTION
**Pull request by aditighag**
* ~#14044 -- test: Bump migrate-svc-test image (@brb)~
 * #14220 -- metricsmap: fix Prometheus exporter (@jibi)
 * #14262 -- docs: Clarify from/toRequires documentation with a new example (@pchaigno)
 * #14264 -- docs: Fix dependency conflict (@joestringer)
 * ~#14153 -- helm/cilium-configmap: added checks to deduplicate keys (@PranaviRoy)~

Once this PR is merged, you can update the PR labels via:
```upstream-prs
$ for pr in 14220 14262 14264; do contrib/backporting/set-labels.py $pr done 1.8; done
```

Note: I had to manually resolve conflicts for `#14153 -- helm/cilium-configmap: added checks to deduplicate keys (@PranaviRoy)`

### CODE REVIEW

**Review by pchaigno — APPROVED** [review]
:+1: for my changes.

**Comment by aditighag** [comment]
test-backport-1.8

**Comment by brb** [comment]
For #14044 needs to be excluded: https://github.com/cilium/cilium/pull/14213#issuecomment-735754326

**Comment by aditighag** [comment]
> #14044 needs to be excluded: [#14213 (comment)](https://github.com/cilium/cilium/pull/14213#issuecomment-735754326)

@brb The PR still has `needs-backport/1.8` label. Does the label need to be removed so that backport scripts don't pick up the change?

**Comment by joestringer** [comment]
@aditighag @brb yes, I've seen this comment across about 3 backporter groups by this point. I think we should nominate someone to take point on pushing those changes back and remove the labels until the underlying problem is resolved.

**Comment by aditighag** [comment]
test-backport-1.8

File: install/kubernetes/cilium/charts/config/templates/configmap.yaml [hunk 1]
```diff
@@ -286,13 +279,18 @@ data:
...
 
-{{- if .Values.global.gke.enabled }}
-
-  ipam: "kubernetes"
+  # Encapsulation mode for communication between nodes
+  # Possible values:
+  #   - disabled
+  #   - vxlan (default)
+  #   - geneve
+{{- if .Values.gke.enabled }}
```
Comment (aanm) [inline install/kubernetes/cilium/charts/config/templates/configmap.yaml]: this should be `Values.global.gke.enabled`

File: install/kubernetes/experimental-install.yaml [hunk 2]
```diff
@@ -104,15 +104,26 @@ data:
...
   #   - geneve
   tunnel: vxlan
+<<<<<<< HEAD
 
   # Name of the cluster. Only relevant when building a mesh of clusters.
   cluster-name: default
+=======
+  # Enables L7 proxy for L7 policy enforcement and visibility
+  enable-l7-proxy: "true"
+>>>>>>> helm/cilium-configmap: added check to deduplicate tunnel and ipam key
```
Comment (aanm) [inline install/kubernetes/experimental-install.yaml]: left-overs?

File: install/kubernetes/quick-install.yaml [hunk 3]
```diff
@@ -90,15 +90,26 @@ data:
...
   #   - geneve
   tunnel: vxlan
+<<<<<<< HEAD
 
   # Name of the cluster. Only relevant when building a mesh of clusters.
   cluster-name: default
+=======
+  # Enables L7 proxy for L7 policy enforcement and visibility
+  enable-l7-proxy: "true"
+>>>>>>> helm/cilium-configmap: added check to deduplicate tunnel and ipam key
```
Comment (aanm) [inline install/kubernetes/quick-install.yaml]: left-overs?

**Review by aanm — CHANGES_REQUESTED** [review]

**Review by jibi — APPROVED** [review]
My changes look good :+1: thanks!

File: install/kubernetes/experimental-install.yaml (hunk 2 above)
Comment (aditighag) [reply install/kubernetes/experimental-install.yaml]: Thanks for catching this. I double checked that I had resolved all the conflicts, probably something went wrong in the 2nd push when I removed one of the changes per last comment. I'll fix it.

File: install/kubernetes/cilium/charts/config/templates/configmap.yaml (hunk 1 above)
Comment (aditighag) [reply install/kubernetes/cilium/charts/config/templates/configmap.yaml]: ~That's not what I see in the PR that was backported - https://github.com/cilium/cilium/pull/14153/files~
Sorry, I get it. It should match the other config names on v1.8. Thanks! Nice catch!

**Comment by aditighag** [comment]
test-backport-1.8

**Review by joestringer — APPROVED** [review]
docs dep fix LGTM :+1:

File: install/kubernetes/cilium/charts/config/templates/configmap.yaml [hunk 4]
```diff
@@ -286,13 +279,18 @@ data:
...
+  #   - geneve
 {{- if .Values.global.gke.enabled }}
-
-  ipam: "kubernetes"
   tunnel: "disabled"
   enable-endpoint-routes: "true"
   blacklist-conflicting-routes: "false"
   enable-local-node-route: "false"
+{{- else }}
+  tunnel: {{ .Values.tunnel }}
```
Comment (joestringer) [inline install/kubernetes/cilium/charts/config/templates/configmap.yaml]:
```suggestion
  tunnel: {{ .Values.global.tunnel }}
```

**Review by joestringer — CHANGES_REQUESTED** [review]

**Review by aanm — APPROVED** [review]

**Comment by joestringer** [comment]
Hi @aditighag , what's the latest status on this PR?

**Comment by aditighag** [comment]
test-backport-1.8

**Comment by aditighag** [comment]
> Hi @aditighag , what's the latest status on this PR?

I pushed the small fix, and restarted tests.

**Comment by aditighag** [comment]
@joestringer Thanks for the nudge. Looks like there are some failures in the quick-install.yaml file that need to be fixed.

**Comment by aditighag** [comment]
test-backport-1.8

**Review by joestringer — APPROVED** [review]
New version of #14153 looks good, same with the backport of the PR I wrote.

**Comment by aditighag** [comment]
test-backport-1.8

**Comment by aditighag** [comment]
test-missed-k8s

**Comment by aditighag** [comment]
test-backport-1.8

**Comment by aditighag** [comment]
>Normal  Pulling    4m    kubelet, k8s2      Pulling image "quay.io/cilium/cilium-builder:2020-12-07-v1.8@sha256:97e32922ef3028d74c2797b3a5fda501be28eafc6499044a01de9cd8d1248940"

Image pulling seems to be slow.

**Comment by aditighag** [comment]
test-missed-k8s

**Comment by aditighag** [comment]
test-missed-k8s

**Comment by aditighag** [comment]
The test (Cilium-PR-Ginkgo-Tests-K8s) failed again  - 
```
"20:56:40 STEP: Ensuring the namespace kube-system exists
time="2020-12-16T20:56:40Z" level=debug msg="running command: kubectl create namespace kube-system"
cmd: "kubectl create namespace kube-system" exitCode: 1 duration: 49.430832ms stdout:
err:
Process exited with status 1
FAIL: failed to ensure the namespace kube-system exists: Process exited with status 1"
```

**Pull request merged**